webviz_subsurface/_datainput/eclipse_init_io/pvt_oil.py

```diff
@@ -112,15 +118,37 @@ def viscosity(self, ratio: np.ndarray, pressure: np.ndarray) -> np.ndarray:
             A list of all viscosity values for the given ratio and pressure values.
 
         """
-        return self.interpolant.viscosity(ratio, pressure)
+        return self.__interpolant.viscosity(ratio, pressure)
+
+    def density(self, ratio: np.ndarray, pressure: np.ndarray) -> np.ndarray:
+        """Args:
+            ratio: List of ratio (key) values the density values are requested for.
+            pressure: List of pressure values the density values are requested for.
+
+        Returns:
+            A list of all density values for the given ratio and pressure values.
+
+        """
+        # rho_o = (rho_o,sc + Rs * rho_g,sc) / B_o
+        fvf_oil = self.formation_volume_factor(ratio, pressure)
+        return np.array(
+            [
+                (
+                    self.__surface_mass_density_oil
+                    + ratio[i] * self.__surface_mass_density_gas
+                )
+                / fvf_oil[i]
+                for i in range(0, len(fvf_oil))
+            ]
```

Comment:
This is a subjective suggestion.
```suggestion
        return np.array(
            [
                (
                    self.__surface_mass_density_oil
                    + ratio_sample * self.__surface_mass_density_gas
                )
                / fvf_oil_sample
                for ratio_sample, fvf_oil_sample in zip(ratio, fvf_oil)
            ]
```